internal/drivers/nomad/driver.go

```diff
@@ -185,39 +185,55 @@ func (p *config) Create(ctx context.Context, opts *types.InstanceCreateOpts) (*t
 	logr.Infoln("scheduler: found a node with available resources")
 
 	// get the machine details where the resource job was allocated
-	ip, id, hostPort, err := p.fetchMachine(logr, resourceJobID)
+	ip, id, hostPorts, err := p.fetchMachine(logr, resourceJobID)
 	if err != nil {
 		defer p.deregisterJob(logr, resourceJobID, false) //nolint:errcheck
 		return nil, err
 	}
+	liteHostEnginePort := hostPorts[vm]
```

Comment:
Let's handle error appropriately in case `hostPorts` does not have en entry to get `liteEngineHostPort`.
You can have the fetchMachine function return lite engine host port separately and gitspaces ports separately.
`ip, id, liteEngineHostPort, gitspacesPorts, err := ...`
Also, do we need hostPorts to be a map here? Can't we just return `gitspacesPorts` as a list of integers?

Reply:
liteEngineHostPort can be returned separately. 
Actually we just need x number of ports. We don't even have to send the list ports to resourceJob() method.
And yes fetchMachine can return just a list of integers. We can then map it appropriately before sending it to the initJob() where we construct the runCmd.